### PR TITLE
Start moving telemetry initialization to service/telemetry

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/5564

### CODE REVIEW

File: service/telemetry/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		t.tracerProvider.Shutdown(ctx),
+	)
+}
+
+// Settings holds configuration for building Extensions.
+type Settings struct {
+	ZapOptions []zap.Option
+}
+
+// New creates a new Extensions from Config.
```
Comment (mx-psi) [inline service/telemetry/telemetry.go:52]:
```suggestion
// New creates a new Telemetry from Config.
```

File: service/telemetry/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+func (t *Telemetry) Shutdown(ctx context.Context) error {
+	// TODO: Sync logger.
+	return multierr.Combine(
+		t.tracerProvider.Shutdown(ctx),
+	)
+}
+
+// Settings holds configuration for building Extensions.
```
Comment (mx-psi) [inline service/telemetry/telemetry.go:47]:
```suggestion
// Settings holds configuration for building Telemetry.
```

**Review by mx-psi — APPROVED** [review]

File: service/telemetry/telemetry.go (hunk 2 above)
Comment (bogdandrutu) [reply service/telemetry/telemetry.go]: Done.

File: service/telemetry/telemetry.go (hunk 1 above)
Comment (bogdandrutu) [reply service/telemetry/telemetry.go]: Done.

**Pull request merged**